### PR TITLE
Inject Dialer from ClusterConfig

### PR DESCRIPTION
**Pull request by rinx**
I'd like to inject net.Dialer from outside of this package.
I added an option to receive net.Dialer in ClusterConfig.

### CODE REVIEW

**Comment by rinx** [comment]
@alourie Could you please review it?

**Comment by rinx** [comment]
@martin-sucha @Zariel @alourie

**Comment by rinx** [comment]
@titanous @alexzorin @dpanth3r @juliendsv @martin-sucha @jacobgreenleaf @zackslash @zhixinwen @furmmon @vrischmann @nathany

**Comment by alourie** [comment]
@rinx is there a specific reason you need this?

**Comment by rinx** [comment]
@alourie Thanks for your response.
We use our own implementation of net.Dialer interface for our products.
The major Golang libraries for RDB/NoSQL (for example go-sql-driver/mysql, go-redis/redis, etc...) provide dialer registration functions/interfaces. This feature makes gocql more useful.

**Comment by martin-sucha** [comment]
Looks good to me.

@alourie There are several possible use-cases for a custom dialer. One is if you need to connect through a proxy (for example over a ssh connection). Another use case is if you need to limit the range of IP addresses the client should connect to, for example if you connect to an address provided by a user and want to exclude certain IP ranges to help mitigate possible server-side request forgery attacks.

File: cluster.go [hunk 1]
```diff
@@ -144,6 +144,9 @@ type ClusterConfig struct {
 	// (default: 200 microseconds)
 	WriteCoalesceWaitTime time.Duration
 
+	// Dialer
```
Comment (martin-sucha) [inline cluster.go]: Maybe the doc comment could be expanded like
```
// Dialer will be used to establish all connections created for this Cluster.
// If not provided, a default dialer configured with ConfigTimeout will be used.
```

**Comment by alourie** [comment]
@martin-sucha that's a fair point.

@rinx it does look good, but I'd ask you to indeed update the comment for the dialer, so it's clear that it's a new option and the default stays as it is. Then I can merge it.

File: cluster.go (hunk 1 above)
Comment (rinx) [reply cluster.go]: Thanks! I've added your comment.

**Comment by rinx** [comment]
@alourie @martin-sucha Thanks for your review. I've just updated the comment for the dialer.

**Comment by alourie** [comment]
While I'm ok with the change, I'll give @Zariel to voice an opinion. If not, I'll merge it tomorrow.

**Comment by alourie** [comment]
@rinx sorry it took a tad longer, but we got there eventually :-)

**Pull request merged**